quantum/via.c

```diff
@@ -529,92 +622,168 @@ void via_qmk_rgblight_set_value(uint8_t *data) {
     }
 }
 
-#endif // #if defined(VIA_QMK_RGBLIGHT_ENABLE)
+void via_qmk_rgblight_save(void) {
+    eeconfig_update_rgblight_current();
+}
 
-#if defined(VIA_QMK_RGB_MATRIX_ENABLE)
+#endif // QMK_RGBLIGHT_ENABLE
+
+#if defined(RGB_MATRIX_ENABLE)
 
 #    if !defined(RGB_MATRIX_MAXIMUM_BRIGHTNESS) || RGB_MATRIX_MAXIMUM_BRIGHTNESS > UINT8_MAX
 #        undef RGB_MATRIX_MAXIMUM_BRIGHTNESS
 #        define RGB_MATRIX_MAXIMUM_BRIGHTNESS UINT8_MAX
 #    endif
 
-// VIA supports only 4 discrete values for effect speed; map these to some
-// useful speed values for RGB Matrix.
-enum speed_values {
-    RGBLIGHT_SPEED_0 = UINT8_MAX / 16, // not 0 to avoid really slow effects
-    RGBLIGHT_SPEED_1 = UINT8_MAX / 4,
-    RGBLIGHT_SPEED_2 = UINT8_MAX / 2,     // matches the default value
-    RGBLIGHT_SPEED_3 = UINT8_MAX / 4 * 3, // UINT8_MAX is really fast
-};
-
-static uint8_t speed_from_rgblight(uint8_t rgblight_speed) {
-    switch (rgblight_speed) {
-        case 0:
-            return RGBLIGHT_SPEED_0;
-        case 1:
-            return RGBLIGHT_SPEED_1;
-        case 2:
-        default:
-            return RGBLIGHT_SPEED_2;
-        case 3:
-            return RGBLIGHT_SPEED_3;
-    }
-}
+void via_qmk_rgb_matrix_command(uint8_t *data, uint8_t length) {
+    // data = [ command_id, channel_id, value_id, value_data ]
+    uint8_t *command_id        = &(data[0]);
+    uint8_t *value_id_and_data = &(data[2]);
 
-static uint8_t speed_to_rgblight(uint8_t rgb_matrix_speed) {
-    if (rgb_matrix_speed < ((RGBLIGHT_SPEED_0 + RGBLIGHT_SPEED_1) / 2)) {
-        return 0;
-    } else if (rgb_matrix_speed < ((RGBLIGHT_SPEED_1 + RGBLIGHT_SPEED_2) / 2)) {
-        return 1;
-    } else if (rgb_matrix_speed < ((RGBLIGHT_SPEED_2 + RGBLIGHT_SPEED_3) / 2)) {
-        return 2;
-    } else {
-        return 3;
+    switch (*command_id) {
+        case id_custom_set_value: {
+            via_qmk_rgb_matrix_set_value(value_id_and_data);
+            break;
+        }
+        case id_custom_get_value: {
+            via_qmk_rgb_matrix_get_value(value_id_and_data);
+            break;
+        }
+        case id_custom_save: {
+            via_qmk_rgb_matrix_save();
+            break;
+        }
+        default: {
+            *command_id = id_unhandled;
+            break;
+        }
     }
 }
 
 void via_qmk_rgb_matrix_get_value(uint8_t *data) {
+    // data = [ value_id, value_data ]
     uint8_t *value_id   = &(data[0]);
     uint8_t *value_data = &(data[1]);
+
     switch (*value_id) {
-        case id_qmk_rgblight_brightness:
+        case id_qmk_rgblight_brightness: {
```

Comment:
Should this be:
```suggestion
        case id_qmk_rgb_matrix_brightness: {
```

Reply:
Thanks, probably happened during the merge.